src/application/controller/UserController.ts

```diff
@@ -0,0 +1,153 @@
+import { BaseController } from './BaseController';
+import { NextFunction, Request, Response } from 'express';
+import { MongoAdapter } from '../../infrastructure/MongoAdapter';
+import { UserRepository } from '../../infrastructure/repository/UserRepository';
+import {operations} from "../../interface/api"
+
+class UserController extends BaseController {
+
+  async getUsers(req: Request, res: Response) {
+    const mongoAdapter = MongoAdapter.getInstance();
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+
+    var query = {};
+
+    // const queryParams = req.query;
+    // for ( const [key, value] of Object.entries(queryParams)) {
+    //   query[key] = value;
+    // }
+    
+    // get all by club
+    if (req.query.club){
+      query["clubMembership.name"] = req.query.club;
+    }
+
+    // get all by university
+    if (req.query.university){
+      query["university"] = req.query.university;
+    }
+
+    // get all by gradlevel
+    if (req.query.gradlevel){
+      query["gradLevel.type"] = req.query.gradlevel;
+    }
+
+    var result = await userRepo.getUsers(query);
+
+    res.status(200).json(result);  
+  }
+
+  async getUserByID(req: Request, res: Response) {
+    var id = req.params.id.toString();
+    const mongoAdapter = MongoAdapter.getInstance();
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+
+    const result = await userRepo.getByID(id);
+
+    if (result != null){
+      res.status(200).json(result);      
+    } else {
+      res.status(404).json();
+    }
+  }
+
+  //generate UUID and pass into request body
+  async createUser(req: Request, res: Response) {
+    const userDetails = req.body;
+    const mongoAdapter = MongoAdapter.getInstance();
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+    userRepo.createUser(userDetails);
+
+    res.status(201).json();
+  }
+
+  async modifyUser(req: Request, res: Response) {
+    const mongoAdapter = MongoAdapter.getInstance();
+
+    const userRepo = new UserRepository(mongoAdapter, 'userDetails');
+    await userRepo.isConnected();
+
+    var id = req.params.id.toString();
+
+    var query = {};
+
+    const queryParams = req.query;
+ /*
```

Comment:
More commented out code to remove

Reply:
possibly useful for developers down the line